### DISCUSSION TITLE
Trying to rescue a condition when previous steps are not run due to the if statement

### DISCUSSION THREAD

**Discussion by y4m4p** [community/community#25548] [Actions]
<p>I want to notify to slack if the first step of the workflow succeeded, or failed.</p>
<p>I have the following workflow, which checks if the 1st step succeeds, and if it does,</p>
<p>it messages via slack with “success”. And if not, it messages via slack with “Something went wrong!”</p>
<pre><code>on:  
 push:  
 branches:  
 - master  
  
jobs:
my-job:
name: Something something
runs-on: ubuntu-latest
env:
SLACK\_OVERRIDE\_MESSAGE: true
SLACK\_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_URL }}
steps:
-name: **This will never run because the ref is always master**
if: github.ref == 'refs/heads/develop'
run: some script
-name: Notify slack when success
if: success()
uses: Ilshidur/action-slack@master
with:
args:"Success!"  
 - name: Notify slack when its not run  
 if: cancelled() \&lt;= This is where I'm having problem with  
 uses: Ilshidur/action-slack@master  
 with:   
 args: "Something went wrong!"
</code></pre>
<p>But to my current knowledge, there is no way to know if the first (or an arbitrary step) has failed in the 3rd step because it can’t backtrack more that 2steps.</p>
<p>So I’m trying to overcome that limitation by chaining the steps using the &lt;if&gt; and &lt;success&gt; and &lt;cancelled&gt; statements.</p>
<p>But when I try this, the 3rd step will never be run, because if the 1st step fails, then the 2nd step’s status would be “NOT RUN(?)” and there is no current way to handle that situation.</p>
<p>I tried using &lt;if: !success()&gt; which resulted with an error, which says</p>
<pre><code>Your workflow file was invalid: The pipeline is not valid. .github/workflows/test.yml: Unexpected tag '!success()'
</code></pre>
<p>If there are any ways to condition a workflow with the success or failure within an arbitrary step, I would be very happy.</p>
<p>Or just any work around for this situation is highly appreciated.</p>
<p>Thank you.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>The job status check functions ‘success(), failure()…’ only check the previous step status in the job. As a workaround, please set the ‘outputs’ for the 1st step, and check the value with ‘if’ in the behind steps. Code sample as below:</p>
<pre><code>jobs:
  my-job:
    name: Something something
    runs-on: ubuntu-latest
    steps:
      - name: set output
        id: step1
        run: |
          echo ::set-output name=status::failure
        continue-on-error: true

      - name: Notify slack when success
        if: steps.step1.outputs.status == 'success'
        run: |
          echo "Success!"
      - name: Notify slack when it not run
        if: steps.step1.outputs.status != 'success'
        run: |
          echo "Something went wrong!"
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181091786-54bf7fcd-9f74-4d15-bdd3-a6354dfc84a3.png" alt="stepstatus.png" data-base62-sha1="tHbG4q9ccWGsoecNLmcIlkrRawM" width="373" height="331"></p>

**Comment by y4m4p** [comment]
<p>Thank you very much weide-zhou!<br>
This worked for me nicely.</p>
